.changeset/shiny-swans-relax.md

```diff
@@ -78,3 +78,46 @@ const viewAlpha = view as TreeViewAlpha<UnsafeUnknownSchema>;
 viewAlpha.initialize([]);
 viewAlpha.root = [];
 ```
+
+Additionally this seems to have negatively impacted co recursive schema which declare an corecursive array as the first schema in the co-recursive cycle.
+Like the TypeScript language our schema system is built on, we don't guarantee exactly which recursive type will compile, but will do our best to ensure useful recursive schema can be created easily.
+In this case a slight change may be required to some recursive schema to get them to compile again:
+
+For example this schema used to compile:
+
+
+```typescript
+class A extends sf.arrayRecursive("A", [() => B]) {}
+{
+	type _check = ValidateRecursiveSchema<typeof A>;
+}
+// Used to work, but breaks in this update.
+class B extends sf.object("B", { x: A }) {}
+```
+
+But now you must use the recursive functions like `objectRecursive` for types which are co-recursive with an array in some cases.
+In our example, it can be fixed as follows:
```

Comment:
FWIW I always just assumed this is how you were supposed to do it anyway. I don't think we made that explicitly clear anywhere but it makes sense, at least to me. The fact that either of the other two ways ever worked or still work feels like a hack to me. So, I don't feel that bad about this "breaking change", but I'm just one data point.

Reply:
Same